types/shares.go

```diff
@@ -57,11 +57,9 @@ func (m Message) MarshalDelimited() ([]byte, error) {
 
 // appendToShares appends raw data as shares.
 // Used for messages.
-func appendToShares(shares []NamespacedShare, id namespace.ID, rawData []byte) []NamespacedShare {
-	nid := make([]byte, len(id))
-	copy(nid, id)
+func appendToShares(shares []NamespacedShare, nid namespace.ID, rawData []byte) []NamespacedShare {
 	if len(rawData) <= MsgShareSize {
-		rawShare := append(nid, rawData...)
+		rawShare := appendCopy(nid, rawData)
```

Comment:
While this method is also nice to read it will be slightly less efficient. See comment below.